smartsim/_core/generation/generator.py

```diff
@@ -249,10 +271,56 @@ def _build_tagged_files(tagged: TaggedFilesHierarchy) -> None:
 
             # write in changes to configurations
             if isinstance(entity, Model):
-                logger.debug(
-                    f"Configuring model {entity.name} with params {entity.params}"
+                files_to_params = self._writer.configure_tagged_model_files(
+                    to_write, entity.params
                 )
-                self._writer.configure_tagged_model_files(to_write, entity.params)
+                self._log_params(entity, files_to_params)
+
+    def _log_params(
+        self, entity: Model, files_to_params: t.Dict[str, t.Dict[str, str]]
+    ) -> None:
+        """Log which files were modified during generation
+
+        and what values were set to the parameters
+
+        :param entity: the model being generated
+        :type entity: Model
+        :param files_to_params: a dict connecting each file to its parameter settings
+        :type files_to_params: t.Dict[str, t.Dict[str, str]]
+        """
+        used_params: t.Dict[str, str] = {}
+        file_to_tables: t.Dict[str, str] = {}
+        for file, params in files_to_params.items():
+            used_params.update(params)
+            table = tabulate([[param, value] for param, value in params.items()],
+                             headers=["Name", "Value"])
+            file_to_tables[relpath(file, self.gen_path)] = table
+
+        if used_params:
+            used_params_str = ", ".join(
+                [f"{name}={value}" for name, value in used_params.items()]
+            )
+            logger.log(
+                level=self.log_level,
+                msg=f"Configured model {entity.name} with params {used_params_str}",
+            )
+            file_table = tabulate(
+                    [[file, table] for file, table in file_to_tables.items()],
+                    headers=["File name", "Parameters"],
+                )
+            log_entry = f"Model name: {entity.name}" + "\n" + file_table + "\n\n"
+            with open(self.log_file, mode="a", encoding="utf-8") as logfile:
+                logfile.write(log_entry)
+            with open(join(entity.path, "param_settings.txt"),
```

Comment:
Incredibly wild corner case to consider here:
```py
exp = Experiment(...)
rs = exp.create_run_settings(...)
ens = experiment.create_ensemble(params=..., run_settings=rs)
ens.attach_generator_files(to_configure=['./param_settings.txt'])
```
will actually overwrite the generated file. I know I was the one who originally asked for this file, but don't like the idea of "banning" users from attaching files of a certain name. Given this corner case, on second thought, I think we should probably only generate the top level `param_settings.txt`

If we do decide to keep the per-entity `param_settings.txt`, we should instead explicitly raise an error if a user attempts to copy/symlink/generate a `param_settings.txt`

Reply:
OK, this needs an in-depth discussion.
The top-level `param_settings.txt` file gets overwritten every time a generation is called. But as we allow partial generation of experiments, this scenario leads to an unexpected result:
```python
exp = Experiment(...)
rs1 = exp.create_run_settings(...)
rs2 = exp.create_run_settings(...)
ens1 = experiment.create_ensemble(params=..., run_settings=rs1)
ens1.attach_generator_files(...)
ens2 = experiment.create_ensemble(params=..., run_settings=rs2)
ens2.attach_generator_files(...)
exp.generate(ens1)  # the top-level file now contains info about ens1 parameters
exp.generate(ens2)  # the top-level file now contains info about ens2 parameters, but ens1 info was overwritten
```
This happens because we are overwriting the file instead of appending.
But if we appended, then there would be no way to overwrite the file if the experiment is re-generated, meaning we would keep appending for every run (even if the user chose `overwrite=True`, because we cannot interpret that correctly again due to partial generation being possible).

I suggest we:
- select a more unique name, such as `smartsim_params.txt` which should be unlikely to be chosen by users
- we make it a reserved file name and `raise ValueError` if the user really wants that file name.


Thoughts?

Reply:
I would be okay with the purposed suggestion above!

Given the reasoning for the replace-not-append decision, I'm inclined to agree with the need for, and am once again in favor of, the per-entity parameter reporting file.

The chances of a name conflict on a `smartsim_params.txt` file seems incredibly unlikely, and if we can raise an exception about a potential file conflict prior to `exp.start(ens)`, that should save users a lot of debugging headache.

Reply:
Added exception at generation level!